src/pages/UpdateServers.js

```diff
@@ -254,6 +254,8 @@ class UpdateServers extends BaseUpdateWizardPage {
   }
 
   async getServerStatuses() {
+    // make sure the model it loaded, we need to fetch some things from it.
+    await this.props.loadModel();
```

Comment:
This still does not solve the race condition -- the call to `this.setState` on line 337 is not guaranteed to complete before the promise is resolved, so line 259 below can still be attempting to reference `this.state.internalModel` before it is set. The conditional `?.` will avoid dereferencing it in that case, so now instead of having a race condition that causes and error we have a race condition that causes no status to be returned.

Reply:
The promise that loadModel now returns does not resolve until react gives us a callback from setState. I didn't quite understand what you were referring to the first go around, this is I think a better solution.